agent/lib_aws_sdk_php.c

```diff
@@ -22,49 +22,43 @@
  * In a normal course of events, the following line will always work
  * zend_eval_string("Aws\\Sdk::VERSION;", &retval, "Get AWS Version")
  * By the time we have detected the existence of the aws-sdk-php and with
- * default composer profject settings, it callable even from
+ * default composer project settings, it is callable even from
  * nr_aws_sdk_php_enable which will automatically load the class if it isn't
- * loaded yet and then evaluate the string. However, in the rare case that files
+ * loaded yet and then evaluate the string. In the rare case that files
  * are not loaded via autoloader and/or have non-default composer classload
- * settings, if the class is not found, PHP 8.2+ will generate a fatal
- * unrecoverable uncatchable error error whenever it cannot find a class. While
- * calling this from nr_aws_sdk_php_enable would have been great and would allow
- * the sdk version value to be set only once, to avoid the very unlikely but not
- * impossible fatal error, this will be called from the
- * "Aws\\ClientResolver::_apply_user_agent" wrapper which GUARANTEES that
- * aws/sdk exists and is already loaded.
- *
- *
- * Additionally given that aws-sdk-php is currently detected from the
- * AwsClient.php file, this method will always be called when a client is
- * created unlike Sdk::construct which doesn't show with PHP 8.2+.
- *
- * Using Aws/Sdk::__construct for version is currently nonviable as it is
- * unreliable as a version determiner.
- * Having separate functionality to extract from Aws/Sdk::__construct
- * is both not required and is redundant and causes additional overhead and
- * so only one function is needed to extract version.
- *
- * Aws\\ClientResolver::_apply_user_agent a reliable function as it is
- * always called on client initialization since it is key to populating
- * the request headers, and it loads Sdk by default.
- *
- * Concerns about future/past proofing to the checking prioritized the following
- * implementation vs using the eval method.
+ * settings, if the class is not found, PHP 8.2+ will generate an
+ * error whenever it cannot find a class which must be caught. Calling this
+ * from nr_aws_sdk_php_enable would allow the sdk version value to be set only
+ * once. To avoid the VERY unlikely but not impossible fatal error, we need to
+ * take an extra step to wrap the call in a try/catch block. This means we need
+ * to have an additionial zend_string_eval to get the result, but we avoid fatal
+ * errors.
```

Comment:
It is possible to use a single zend_eval_string to get the sdk version with error handling. Take a look at https://github.com/newrelic/newrelic-php-agent/blob/R11.4/agent/lib_composer.c#L91. That code defines a lambda (anonymous function) and calls it in place. The same idea can be used for the AWS SDK version retrieval. The PHP code to be used in a single `zend_string_eval` could look like this:
```php
(function() {
     $nr_aws_sdk_version = '';
     try {
          $nr_aws_sdk_version = Aws\\Sdk::VERSION;
     } catch (Throwable $e) {
     }
     return $nr_aws_sdk_version;
})();
```